File: Java/ExtremeTournament/src/extreme/model/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package extreme.model;

/**
 *
 * @author dell
 */
public class Produit {
    
    private int refProd;
    private String nomProd;
    private float prix;
    private int TotalEnStock;
    private String Descriptif;
    private String CategorieProd;
    private String disponibilite;
    
    public Produit () {    
    }

    public Produit(int refProd, String nomProd, float prix, int TotalEnStock, String Descriptif, String CategorieProd, String disponibilite) {
        this.refProd = refProd;
        this.nomProd = nomProd;
        this.prix = prix;
        this.TotalEnStock = TotalEnStock;
        this.Descriptif = Descriptif;
        this.CategorieProd = CategorieProd;
        this.disponibilite = disponibilite;
    }

    public Produit(String nomProd, float prix, int TotalEnStock, String Descriptif, String CategorieProd, String disponibilite) {
        this.nomProd = nomProd;
        this.prix = prix;
        this.TotalEnStock = TotalEnStock;
        this.Descriptif = Descriptif;
        this.CategorieProd = CategorieProd;
        this.disponibilite = disponibilite;
    }
    

    public int getRefProd() {
        return refProd;
    }

    public String getNomProd() {
        return nomProd;
    }

    public float getPrix() {
        return prix;
    }

    public int getTotalEnStock() {
        return TotalEnStock;
    }

    public String getDescriptif() {
        return Descriptif;
    }

    public String getCategorieProd() {
        return CategorieProd;
    }

    public String getDisponibilite() {
        return disponibilite;
    }

    public void setRefProd(int refProd) {
        this.refProd = refProd;
    }

    public void setNomProd(String nomProd) {
        this.nomProd = nomProd;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public void setTotalEnStock(int TotalEnStock) {
        this.TotalEnStock = TotalEnStock;
    }

    public void setDescriptif(String Descriptif) {
        this.Descriptif = Descriptif;
    }

    public void setCategorieProd(String CategorieProd) {
        this.CategorieProd = CategorieProd;
    }

    public void setDisponibilite(String disponibilite) {
        this.disponibilite = disponibilite;
    }

    @Override
    public String toString() {
        return "Produit{" + "refProd=" + refProd + ", nomProd=" + nomProd + ", prix=" + prix + ", TotalEnStock=" + TotalEnStock + ", Descriptif=" + Descriptif + ", CategorieProd=" + CategorieProd + ", disponibilite=" + disponibilite + '}';
    }
    

}

File: ExtremeTournament/src/Entity/Produit.php
<?php

namespace App\Entity;

use App\Repository\ProduitRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=ProduitRepository::class)
 */
class Produit
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $refProd;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $nomProd;

    /**
     * @ORM\Column(type="float")
     */
    private $prix;

    /**
     * @ORM\Column(type="integer")
     */
    private $TotalEnStock;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $Descriptif;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $CategorieProd;

    /**
     * @ORM\Column(type="string", length=50)
     */
    private $disponibilite;

    /**
     * @ORM\ManyToOne(targetEntity=Panier::class, inversedBy="refProd")
     * @ORM\JoinColumn(name="id_panier",referencedColumnName="id_panier",nullable=false)
     */
    private $panier;


    public function getRefProd(): ?int
    {
        return $this->refProd;
    }

    public function setRefProd(int $refProd): self
    {
        $this->refProd = $refProd;

        return $this;
    }

    public function getNomProd(): ?string
    {
        return $this->nomProd;
    }

    public function setNomProd(string $nomProd): self
    {
        $this->nomProd = $nomProd;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getTotalEnStock(): ?int
    {
        return $this->TotalEnStock;
    }

    public function setTotalEnStock(int $TotalEnStock): self
    {
        $this->TotalEnStock = $TotalEnStock;

        return $this;
    }

    public function getDescriptif(): ?string
    {
        return $this->Descriptif;
    }

    public function setDescriptif(string $Descriptif): self
    {
        $this->Descriptif = $Descriptif;

        return $this;
    }

    public function getCategorieProd(): ?string
    {
        return $this->CategorieProd;
    }

    public function setCategorieProd(string $CategorieProd): self
    {
        $this->CategorieProd = $CategorieProd;

        return $this;
    }

    public function getDisponibilite(): ?string
    {
        return $this->disponibilite;
    }

    public function setDisponibilite(string $disponibilite): self
    {
        $this->disponibilite = $disponibilite;

        return $this;
    }

    public function getPanier(): ?Panier
    {
        return $this->panier;
    }

    public function setPanier(?Panier $panier): self
    {
        $this->panier = $panier;

        return $this;
    }
}
